### PR TITLE
Remove lateinit usage in Main.kt

### LINKED ISSUE #132: Remove lateinit in project property in Main class
After M12.2 is out, LateinitUsage will report a code smell in Main class.

Idea: Refactor `lateinit project` to `var project: Path = null` and validate notNull in `validateCli`. (Consider backing field if kontlinc want to many !!'s)

### PR DESCRIPTION
**Pull request by Mauin**
Resolves #132 

This follows the recommendation from the issue. But overall I'm not sure if its nicer to have 3 `!!` in the code compared to the `lateinit var`.

Probably also the null check in `validateCli` is not really useful as the project parameter is already `required = true` and thus JCommander will already throw an exception if the parameter is not specified.

### CODE REVIEW

**Comment by arturbosch** [comment]
Yeah that is not pretty at all :), my other thought was to introduce a backing field so we do not need the `!!` everywhere?

**Comment by Mauin** [comment]
Yep, agree. I'll explore some other approaches later today. 👍

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt [hunk 1]
```diff
@@ -96,6 +99,9 @@ class Main {
 		private fun validateCli(cli: Main): List<String> {
 			val violations = ArrayList<String>()
 			with(cli) {
+				if (project == null) {
```
Comment (Mauin) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt]: As mentioned this is not really necessary as `JCommander` already requires this parameter to be set. Let me know how you feel about this one also being here.

**Comment by Mauin** [comment]
Updated the PR by using the backing field and making `project` private to prefer using the non-nullable field. Looks better for me.

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt (hunk 1 above)
Comment (arturbosch) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt]: Then delete this lines and do a test run with the CLI and we are happy :)

**Comment by MyDogTom** [comment]
I'm just curious why `lateinit` is a bad practice?  Especially when it's replaced with `!!`. For me this change means that `UninitializedPropertyAccessException` will be just replaced with `NullPointerException`.

**Comment by arturbosch** [comment]
Because it misleads you not using a safer way to initialize a variable (constructor injection or delegation).

When I started programming in Kotlin I often used `lateinit` in spring-framework or just because I found using constructors in Kotlin difficult (always calling primary constructor in secondary constructors etc) and I think this is a common pitfall for newcomers. 

Ofc in our case it seems kind of odd to replace one keyword with a backing field, but this due to the nature of many annotation heavy libraries (jcommander). Ofc if you know what you do `lateinit` is not that bad and in our solution we are also relying on jcommander to check the field, while the backing field looks unsafe ...

The tradeoff here is like Suppress annotation vs backing field :D

**Pull request merged**